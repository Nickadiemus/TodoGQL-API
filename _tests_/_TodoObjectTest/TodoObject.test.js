//requires TodoObject.js modules
const {FetchTodoObjects, FetchSingleTodoObject } = require('./TodoObject')

//test id: must be changed if ran on another instance due to local DB
var id = "TtqWyeRJLbC67Hs7G";

/*/
 *  Type of Test: Query Snapshot
 *  Proves: graphql Endpoint returns actual data equal to a snapshot
 *  @param(s): none
 *  Description: an asynchronous call to the localhost:3000/graphql Endpoint
 *  that will verify by a snapshot the query matches the response given

 NOTE: TodoObjects must be populated before this test can be ran
/*/

test('test call for TodoObjects Request', async () => {
  const data = await FetchTodoObjects();
  expect(data).toMatchSnapshot();
});

/*/
 *  Type of Test: Query Snapshot
 *  Proves: graphql Endpoint returns actual data equal to a snapshot
 *  @param(s): id
 *  Description: an asynchronous call to the localhost:3000/graphql Endpoint
 *  that will verify a single TodoObject by a snapshot the query matches the response given

 NOTE: TodoObjects must be populated before this test can be ran and an id must
 be provided
/*/
test('test call for FetchSingleTodoObject Request', async () => {
  const data = await FetchSingleTodoObject(id);
  expect(data).toMatchSnapshot();
})

/*/
 *  Type of Test: Unit Testing
 *  Proves: data returned from API are corresponding to defined schemas
 *  @param(s): none
 *  Description: an asynchronous call to the localhost:3000/graphql Endpoint
 *  that will unit test every single expected unit of TodoObject and its TodoItems

 NOTE: TodoObjects must be populated before this test can be ran
/*/

it('Unit Test for all TodoObjects/TodoItems: no empty TodoObjects', async () => {
  const data = await FetchTodoObjects();
  count = 0;
  data.data.TodoObjects.forEach(item => {
    expect(typeof(item._id)).toBe('string');                //expects item._id to be a String Type
    expect(typeof(item.name)).toBe('string');               //expects item.name to be a String Type
    expect(typeof(item.completed)).toBe('boolean');         //expects item.completed to be a Boolean Type
    expect(typeof(item.items)).toBe('object');              //expects item.items to be a Array Type
    count++
    item.items.forEach(subItem => {
      expect(typeof(subItem._id)).toBe('string')            //expects subItem._id to be a String Type
      expect(typeof(subItem.parent)).toBe('string')         //expects subItem.parent to be a String Type
      expect(typeof(subItem.content)).toBe('string')        //expects subItem.content to be a String Type
      expect(typeof(subItem.completed)).toBe('boolean')     //expects subItem.completed to be a String Type
      count++
    })
  })
  console.log("Loop(s): " + count);
})
