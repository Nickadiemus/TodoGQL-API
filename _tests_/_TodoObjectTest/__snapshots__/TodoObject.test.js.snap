// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test call for FetchSingleTodoObject Request 1`] = `
Object {
  "data": Object {
    "FindTodoObject": Object {
      "_id": "TtqWyeRJLbC67Hs7G",
      "completed": false,
      "items": Array [
        Object {
          "_id": "mb4zcsEbGwzA6XuTz",
          "completed": false,
          "content": "TodoItem1",
          "parent": "TtqWyeRJLbC67Hs7G",
        },
        Object {
          "_id": "xavRNCd5RJfhTdJjK",
          "completed": false,
          "content": "TodoItem2",
          "parent": "TtqWyeRJLbC67Hs7G",
        },
      ],
      "name": "TodoObject1",
    },
  },
}
`;

exports[`test call for TodoObjects Request 1`] = `
Object {
  "data": Object {
    "TodoObjects": Array [
      Object {
        "_id": "TtqWyeRJLbC67Hs7G",
        "completed": false,
        "items": Array [
          Object {
            "_id": "mb4zcsEbGwzA6XuTz",
            "completed": false,
            "content": "TodoItem1",
            "parent": "TtqWyeRJLbC67Hs7G",
          },
          Object {
            "_id": "xavRNCd5RJfhTdJjK",
            "completed": false,
            "content": "TodoItem2",
            "parent": "TtqWyeRJLbC67Hs7G",
          },
        ],
        "name": "TodoObject1",
      },
      Object {
        "_id": "DeFkpyTASdvxGnSmX",
        "completed": false,
        "items": Array [
          Object {
            "_id": "7W6dsBuZpRP7233mE",
            "completed": false,
            "content": "TestItem1",
            "parent": "DeFkpyTASdvxGnSmX",
          },
          Object {
            "_id": "vXMt2uzo8rbheR7ju",
            "completed": false,
            "content": "TestItem2",
            "parent": "DeFkpyTASdvxGnSmX",
          },
          Object {
            "_id": "bkgAg6PbHeeGGZnpZ",
            "completed": false,
            "content": "TestItem3",
            "parent": "DeFkpyTASdvxGnSmX",
          },
        ],
        "name": "TodoObject2",
      },
      Object {
        "_id": "shQMsTn2uHFzAYBmg",
        "completed": false,
        "items": Array [
          Object {
            "_id": "GFsG62dTm2MYob6d3",
            "completed": false,
            "content": "TestItem1",
            "parent": "shQMsTn2uHFzAYBmg",
          },
        ],
        "name": "TodoObject3",
      },
    ],
  },
}
`;
